# 要約 
## 要約

このディスカッションは、Kaggleコンペティション「LLM 20 Questions」における回答者エージェントのバグに関するものです。

**問題:**

* 回答者エージェントは最初のラウンドでキーワードを受け取っていない。
* リプレイログによると、回答者エージェントは空のキーワード/カテゴリ値でアクティブになっている。
* エージェントは最初のラウンドで簡単な質問を「幻覚」しているように見える。
* この問題は、スターターノートブックのfstringがobs.keywordプロパティを盲目的にアクセスしているため、隠されている。

**原因:**

* スターターノートブックのfstringがobs.keywordプロパティを盲目的にアクセスしているため、空の文字列が渡されている。

**影響:**

* 回答者エージェントは最初のラウンドで適切な情報を受け取ることができない。
* エージェントは誤った回答を生成する可能性がある。

**解決策:**

* 主催者はこの問題を確認する必要がある。
* ユーザーは、回答者エージェントにキーワードが渡されていない場合にエラーを発生させるテストコードを追加することができる。

**主催者の対応:**

* Bovard Doerschuk-Tiberiは、この問題を確認するとコメントしている。


---
# バグ: 回答者エージェントに最初のラウンドでキーワードが渡されていない

**monoxgas** *2024年5月22日 水曜日 10:16:36 日本標準時* (7票)

これは主催者によって確認されるべきですが、回答者エージェントが初めて使用された際に、その観察結果にキーワードが渡されていないことはほぼ確実です。

リプレイログを調べると、回答者エージェントがステップ2で空のキーワード/カテゴリ値でアクティブになっていることがわかります。また、エージェントは最初のラウンドで頻繁に簡単な質問を「幻覚」しているように見えます。この問題は、スターターノートブックのfstringが単にobs.keywordプロパティを盲目的にアクセスしているため、隠されています。これは空の文字列です。

また、この状況が発生した場合にエラーを発生させるテストコードを提出物に追加しましたが、検証中に例外が発生します。

```python
def answer(base: rg.PendingChat, observation: Observation) -> t.Literal["yes", "no"]:
    if not observation.keyword:
        print("Keyword wasn't provided to answerer", file=sys.stderr)
        raise Exception("Keyword wasn't provided to answerer")
```

例外が発生しました。
回答者エージェントに最初の質問が渡されました。
キーワードが利用できない状態で「yes」を選択しました。
---
# 他のユーザーからのコメント

> ## Bovard Doerschuk-Tiberi
> 
> ご報告ありがとうございます。確認してみます！
> 
> 
> 
--- 

